compute_tools/src/installed_extensions.rs

```diff
@@ -72,9 +81,97 @@ pub async fn get_installed_extensions(connstr: Url) -> Result<InstalledExtension
             }
         }
 
-        Ok(InstalledExtensions {
+        let res = InstalledExtensions {
             extensions: extensions_map.values().cloned().collect(),
-        })
+        };
+
+        // set the prometheus metrics
+        for ext in res.extensions.iter() {
+            let versions = {
+                let mut vec: Vec<_> = ext.versions.iter().cloned().collect();
+                vec.sort();
+                vec.iter()
+                    .map(|x| x.to_string())
+                    .collect::<Vec<_>>()
+                    .join(",")
+            };
+
+            INSTALLED_EXTENSIONS
+                .with_label_values(&[
+                    &tenant_id.to_string(),
+                    &timeline_id.to_string(),
+                    &ext.extname,
+                    &versions,
+                ])
+                .set(ext.n_databases as u64);
+        }
+
+        Ok(res)
     })
     .await?
 }
+
+static INSTALLED_EXTENSIONS: Lazy<UIntGaugeVec> = Lazy::new(|| {
+    register_uint_gauge_vec!(
+        "installed_extensions",
+        "Number of databases where extension is installed, versions passed as label",
+        &["tenant_id", "timeline_id", "extension_name", "versions"]
```

Comment:
Hm, my first reaction is that using Prom metrics for this seems to be a wrong tool.

We basically add a generic gauge, where the value is roughly binary -- we only care if something is installed or not, not then number of DBs inside project/timeline.

Yet, then we put all the valuable info inside labels. That way we explode the number of timeseries pretty quickly -- N_tenants*N_timelines*N_extensions*N_historical_versions_we_had_within_retention_window

It could be that I miss something and that's an OKish use-case, but I'd get an explicit buy in from the SRE -- @lassizci @mickael-carl

Reply:
FWIW, I am pretty sure that 99% of users use only default version of extensions.

If we have another way to identify the source of the metrics (pod_id? compute_id?), i will gladly remove redundant `tenant_id`, `timeline_id` labels.

Reply:
It's indeed not the best use case for prometheus, but will do as there isn't anything better.

As for `version`, a comma separated multi-value label isn't very ergonomic to query - it would be better to produce metric per value.

As for `tenant_id` / `timeline_id` - at which level are we interested in the metrics? We can configure <compute/project|etc>_id to be set at scrape time.

Reply:
> FWIW, I am pretty sure that 99% of users use only default version of extensions.

Yeah, I meant more that default will change over time, and people will likely do upgrades rarely. So we should end up with roughly all historical versions. But we will see this after implementing this analytics :)

Reply:
>As for version, a comma separated multi-value label isn't very ergonomic to query - it would be better to produce metric per value.

for now we only collect total number of databases, not per-version. so splitting this line won't make a lot of sense.
i.e. version="1.0,1.1" -> 5 means that we have
extension installed in 5 databases, in some with versions 1.0 in some with version 1.1.

@oruen, wdyt? Is this format ok for your data collection purposes?

>As for tenant_id / timeline_id - at which level are we interested in the metrics? We can configure <compute/project|etc>_id to be set at scrape time.

on per-timeline (or branch_id) level.
For most cases per-project will be enough, but I think there can be cases with different extensions in different branches. @ololobus , wdyt?

Reply:
Re: "1.0,1.1" — that will work just fine as long as there are no commas in version strings.

If branch_id/project_id can be added cheaply as labels to the metric, it would make using it easier. If not, it is also fine.

Reply:
Is the ordering of the versions somehow quaranteed? Because if it's not, that will result in unnecessary cardinality.

Reply:
>Is the ordering of the versions somehow guaranteed? Because if it's not, that will result in unnecessary cardinality, as 1.0,1.1 != 1.1,1.0 from storage perspective.

yes, I sort them before putting into label.
https://github.com/neondatabase/neon/pull/9375/files#diff-a91edc65af5e471002cbef8d2ea529c46f7afb7487c95e8f99238f3867346034R92

Reply:
@lassizci , I can remove tenant/timeline labels, because they are redundant, right?

Reply:
We haveallof those available if needed, so no need for those in here, I thin.

Reply:
fixed